src/main/java/com/booking/tickets/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.booking.tickets.security;
+
+import com.booking.tickets.exception.AuthenticationException;
+import com.booking.tickets.exception.DataProcessingException;
+import com.booking.tickets.lib.Inject;
+import com.booking.tickets.lib.Service;
+import com.booking.tickets.model.User;
+import com.booking.tickets.service.UserService;
+import com.booking.tickets.util.HashUtil;
+import javax.persistence.NoResultException;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user;
+        try {
+            user = userService.findByEmail(email);
+        } catch (DataProcessingException e) {
+            if (e.getCause().getClass() == NoResultException.class) {
+                throw new AuthenticationException("This login does not exist");
+            }
+            throw e;
```

Comment:
let's remove this check and just pass exception to constructor

Reply:
Optional: look closely to uniqueResult() method for query. See the difference between it and getSingleResult(), probably it will be more suitable.

Reply:
thank you! I decided to use uniqueResult() in UserDaoImpl, now I don't need to catch DataProcessingException in AuthenticationServiceImpl